### PR TITLE
Add a method to create a document with an auto-generated id to `Store`

### PR DESCRIPTION
**Pull request by matt989253**

### CODE REVIEW

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/infrastructure/persistence/store/fake/FakeCollectionReference.kt [hunk 1]
```diff
@@ -24,6 +26,14 @@ class FakeCollectionReference : CollectionReference, DocumentBuilder, FakeQuery
 
   private val state = MutableStateFlow(State())
 
+  override fun document(): DocumentReference {
+    val id = UUID.randomUUID().toString()
+    return state.updateAndGetWithValue {
+      val (doc, ref) = it.documents.getOrPut(id) { FakeDocumentReference(FakeDocumentId(id)) }
+      it.copy(documents = doc) to ref
+    }
```
Comment (alexandrepiveteau) [inline mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/infrastructure/persistence/store/fake/FakeCollectionReference.kt:34]:
```suggestion
    return document(UUID.randomUUID().toString())
```

File: mobile/src/main/java/ch/epfl/sdp/mobile/infrastructure/persistence/store/CollectionReference.kt [hunk 2]
```diff
@@ -8,6 +8,13 @@ import kotlinx.coroutines.flow.Flow
  */
 interface CollectionReference : Query {
 
+  /**
+   * Accesses (creates) a document with no specified path (generates one automatically).
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/infrastructure/persistence/store/CollectionReference.kt:12]: I'd rather say something along the lines of "Creates a document with a unique identifier."

File: mobile/src/main/java/ch/epfl/sdp/mobile/infrastructure/persistence/store/firestore/FirestoreCollectionReference.kt [hunk 3]
```diff
@@ -14,6 +14,7 @@ import com.google.firebase.firestore.CollectionReference as ActualCollectionRefe
 class FirestoreCollectionReference(
     private val reference: ActualCollectionReference,
 ) : CollectionReference, Query by FirestoreQuery(reference) {
+  override fun document(): DocumentReference = FirestoreDocumentReference(reference.document())
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/infrastructure/persistence/store/firestore/FirestoreCollectionReference.kt:17]: Please add a line above.

**Review by alexandrepiveteau — DISMISSED** [review]
Looking good to me, thanks !

**Review by alexandrepiveteau — APPROVED** [review]

**Review by KurohanaJuri — APPROVED** [review]
LGTM

**Pull request merged**